yapapi/executor/__init__.py

```diff
@@ -368,6 +368,32 @@ async def accept_payment_for_agreement(agreement_id: str, *, partial: bool = Fal
             )
 
         async def find_offers() -> None:
+            async def reject_proposal(proposal, reason):
```

Comment:
hmm... the only thing I'm concerned about is that again, we're adding inner-scope functions ... whereas maybe we should start pulling that stuff outside ... but I do understand that the milk is spilled by `find_offers` already so it would be artificial to pull these as methods and leave `find_offers` as internal...

Reply:
Sometimes inner functions are just better - object-oriented programming is not the only paradigm, let's not be afraid of them. Actually, I like @azawlocki's solution very much, it's an improvement over the original code and over my PR to `yajsapi`, which AFAIR added another copy of try/call api/catch/emit.

Reply:
These functions are not visible outside of `find_offers()`. If they are not used outside, IMO they shouldn't be moved as they are "helper" functions.

Reply:
@shadeofblue  I've just started pulling stuff outside in #306, I'm not sure we want to do that kind of refactoring in a PR to a release branch.